examples/cg_possion.py

```diff
@@ -0,0 +1,60 @@
+import taichi as ti
+import math
+ti.init(ti.cuda)
+
+n = 512
+steps = 32
+eps = 1e-5
+
+b = ti.field(ti.f32, (n, n))
+x = ti.field(ti.f32, (n, n))
+d = ti.field(ti.f32, (n, n))
+r = ti.field(ti.f32, (n, n))
+
+
+@ti.func
+def c(x: ti.template(), i, j):
+    return x[i, j] if 0 <= i < n and 0 <= j < n else 0.0
+
+
+@ti.func
+def A(x: ti.template(), I):
+    i, j = I
+    return x[i, j] * 4 - c(x, i - 1, j) - c(x, i + 1, j) \
+            - c(x, i, j - 1) - c(x, i, j + 1)
+
+
+@ti.kernel
+def init():
+    for i in ti.grouped(x):
+        d[i] = b[i] - A(x, i)
+        r[i] = d[i]
+
+
+@ti.kernel
+def substep():
+    alpha, beta, dAd = eps, eps, eps
```

Comment:
`beta` doesn't have to start with `eps` initially. Please initialize these to zero and add `eps` only when they are used as denominators.